Vostok.ServiceDiscovery/ServiceBeacon.cs

```diff
@@ -112,11 +119,19 @@ public void Stop()
                     checkNodeSignal.Set();
                     beaconTask.GetAwaiter().GetResult();
                     stopCancellationToken.Dispose();
-                    if (!DeleteNodeAsync().GetAwaiter().GetResult())
-                    {
-                        Task.Run(DeleteNodeTask);
-                    }
                     RemoveTagsIfNeed().GetAwaiter().GetResult();
+
+                    DeleteResult deleteResult;
+                    if ((deleteResult = DeleteNodeAsync().GetAwaiter().GetResult()).IsSuccessful)
+                        SendEvent(ServiceDiscoveryEventKind.ReplicaStopped,
+                            deleteResult.Status == ZooKeeperStatus.Ok ? "Replica unregistered, because Service Beacon stopped." : "Beacon stop, but replica unregistered before.");

```

Comment:
```suggestion
                            deleteResult.Status == ZooKeeperStatus.Ok ? "Replica unregistered, because Service Beacon stopped." : "Beacon stopped, but replica has unregistered before.");

```